sympy/functions/combinatorial/numbers.py

```diff
@@ -2292,3 +2292,149 @@ def ok(x):
         return Integer(sum(1 for i in multiset_derangements(s)))
     return Integer(abs(integrate(exp(-x)*Mul(*[
         laguerre(i, x)**m for i, m in counts.items()]), (x, 0, oo))))
+
+
+#----------------------------------------------------------------------------#
+#                                                                            #
+#                           Padovan numbers                                  #
+#                                                                            #
+#----------------------------------------------------------------------------#
+
+
+class padovan(Function):
+    r"""
+    In number theory, the Padovan sequence is the sequence of integers P(n)
+    defined by the initial values `P_0 = 1`, `P_1 = 1` , `P_2 = 1` and the
+    the two-term recurrence `P_n = P_{n-2} + P_{n-3}`.
+
+    * ``padovan(n)`` gives the `n^{th}` Padovan number, `P_n`
+
+    Examples
+    ========
+
+    >>> from sympy import padovan
+
+    >>> padovan.is_padovan(2)
+    True
+    >>> [padovan(x) for x in range(0,11)]
+    [1, 1, 1, 2, 2, 3, 4, 5, 7, 9, 12]
+    >>> padovan.find_first_n_padovan_numbers(8)
+    [1, 1, 1, 2, 2, 3, 4, 5]
+    >>> padovan.find_padovan_numbers_in_range(2,39)
+    [2, 2, 3, 4, 5, 7, 9, 12, 16, 21, 28, 37]
+
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Padovan_sequence
+    .. [2] https://mathworld.wolfram.com/PadovanSequence.html
+
+    """
+
+    @staticmethod
+    def is_padovan(n):
+        try:
+            n = as_int(n)
+        except ValueError:
+            return False
+        if n>0:
+            if n == 1 :
+                return True
+
+            sn0 = 1
+            sn1 = 1
+            sn2 = 1
+            while sn2 < n :
+                b = sn0 + sn1
+                sn0 = sn1
+                sn1 = sn2
+                sn2 = b
+
+            if Eq(sn2, n, evaluate=True):
+                return True
+            else:
+                return False
```

Comment:
Both `sn2` and `n` are of type `int` if I understand this correctly. In that case you just do `return sn2 == n`.

Reply:
> Both `sn2` and `n` are of type `int` if I understand this correctly. In that case you just do `return sn2 == n`.

I will remove unnecessary code in a future commit !